### PR TITLE
UNR-1404 Schema / snapshot missing check before local deployment start.

### PR DESCRIPTION
**Pull request by bartoszrudzki-improbable**
#### Description
I added checks if several file exists when user tries to start local deployment. If not I print error to console to generate schema / snapshot - dependent what is missing. 

#### Release note
Feature: Schema and snapshot missing checks before local deployment start. In case of error proper log will be produced.

#### Tests
I deleted each file, run game and check if error is printed.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include "SpatialGDKServicesModule.h"
 #include "SpatialGDKSettings.h"
 #include "SpatialGDKSimulatedPlayerDeployment.h"
+#include "SpatialGDKEditorSchemaGenerator.h"
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: Please alphabetize these includes

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp [hunk 2]
```diff
@@ -811,6 +824,19 @@ void FSpatialGDKEditorToolbarModule::GenerateSchema(bool bFullScan)
 	}
 }
 
+bool FSpatialGDKEditorToolbarModule::IsSnapshotGenerated(FString snapshotName) const
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]:
```suggestion
bool FSpatialGDKEditorToolbarModule::IsSnapshotGenerated(const FString& snapshotName) const
```

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp [hunk 3]
```diff
@@ -547,6 +548,18 @@ void FSpatialGDKEditorToolbarModule::VerifyAndStartDeployment()
 		return;
 	}
 
+	if (!IsSnapshotGenerated())
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: Bit of a driveby - but seems reasonable that we should be using the GetSpatialOSSnapshotFile() both here, and when we're actually launching the deployment. Either that, or add a `snapshot to save` and `snapshot to load` setting to our editor settings.

Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: I'm behind this idea of using `GetSpatialOSSnapshotFile()`, although adding the `snapshot to load` and `snapshot to save` is a bit of scope creep. What do you think @bartoszrudzki-improbable

Comment (bartoszrudzki-improbable) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: As I check file existence by full path I will use GetSnapshotPath() method.

Comment (bartoszrudzki-improbable) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: @joshuahuburn if we want `snapshot to load` and `snapshot to save` functionality I think we should create other ticket for this.

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: Only worry there is if a user actually sets that within the editor, we'd be checking against that name, and still launching with the `default.snapshot` name.

Comment (bartoszrudzki-improbable) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: Oh this is not intuitive at all. Why don't we use snapshot name from settings during launch?

Comment (bartoszrudzki-improbable) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: I hard coded path for default snapshot right now. As I said previously if we want `snapshot to load` and `snapshot to save` functionalities, I think it should be done in a different ticket as it requires some additional work with spot.exe (to force using specified snapshot file).

Comment (bartoszrudzki-improbable) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp]: After talk with @joshuahuburn I implemented `snapshot to load` and `snapshot to save` solution.

File: SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp [hunk 4]
```diff
@@ -156,7 +156,8 @@ bool FLocalDeploymentManager::TryStartLocalDeployment(FString LaunchConfig, FStr
 		TryStartSpatialService();
 	}
 
-	FString SpotCreateArgs = FString::Printf(TEXT("alpha deployment create --launch-config=\"%s\" --name=localdeployment --project-name=%s --json %s"), *LaunchConfig, *FSpatialGDKServicesModule::GetProjectName(), *LaunchArgs);
+	SnapshotName.RemoveFromEnd(".snapshot");
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp]:
```suggestion
	SnapshotName.RemoveFromEnd(TEXT(".snapshot"));
```

**Comment by m-samiec** [comment]
Can you please add a release note before merging - @joshuahuburn to sign off.

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Features:
 - Visual Studio 2019 is now supported.
+- Schema and snapshot missing checks before local deployment start. In case of error proper log will be produced.
```
Comment (joshuahuburn) [inline CHANGELOG.md]:
```suggestion
- Added a a check for schema and snapshot before attempting to start a local deployment. If either are missing then an error message will be displayed.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Features:
 - Visual Studio 2019 is now supported.
+- Added a a check for schema and snapshot before attempting to start a local deployment. If either are missing then an error message will be displayed.
```
Comment (m-samiec) [inline CHANGELOG.md]:
```suggestion
- Added a check for schema and snapshot before attempting to start a local deployment. If either are missing then an error message will be displayed.
```
